Source/WebCore/platform/ios/PlaybackSessionInterfaceAVKit.mm

```diff
@@ -84,9 +82,19 @@
     invalidate();
 }
 
-PlaybackSessionModel* PlaybackSessionInterfaceAVKit::playbackSessionModel() const
+WebAVPlayerController *PlaybackSessionInterfaceAVKit::playerController() const
+{
+    return m_playerController.get();
+}
+
+void PlaybackSessionInterfaceAVKit::invalidate()
 {
-    return m_playbackSessionModel;
+    if (!m_playbackSessionModel)
+        return;
+
+    [m_playerController setDelegate:nullptr];
+    m_playbackSessionModel->removeClient(*this);
+    m_playbackSessionModel = nullptr;
```

Comment:
I think the last two lines here should be moved to `PlaybackSessionInterfaceIOS::invalidate()` since both subclasses need to do this.